### ISSUE TITLE
Company completion does not work with some projects

### ISSUE THREAD

**Issue by chep** [Andersbakken/rtags#451]
Hi Anders,

I have a strange bug with company. I have several projects on my machine. It
works well for most of them but not for others (the more complicated ones).

rtags-find-symbol-at-point and rtags-display-summary work for all projects even
when company fails.

When I create a simple project with only this:
```c
struct Plop
{
    int lalala;
};


int main()
{
    Plop p;

    return 0;
}
```
It works. When I type "p." company will display "lalala" wich is correct.

If I add this simple structure in a cpp file of a "failing" project and ask for
completion, company says "No completion found".

I don't know how to debug this. I tried to enable rc logs and I see the same
commands for both projects (except for myFile.cpp of course):

/home/chep/bin/rc --current-file=myFile.cpp --unsaved-file=myFile.cpp:54803 -Y -l myFile.cpp:125:9:

When I try this command in a terminal, it never returns(for both projects), it
seems to work only in Emacs.

In Emacs all I see is a "timeout" in company-rtags--candidates:
```lisp
(while (and (eq old rtags-last-completions)
			(> maxwait 0))
	(decf maxwait)
	(sleep-for company-async-wait))
```
I don't know how to reproduce the bug, it never happens when I try to create a
simple test project but I can try to debug it. The problem is that I don't know
where to start. Any idea?

Thanks.

**Comment by Andersbakken** [comment]
Hi

It's kinda tricky to debug actually. Your best bet might be to start adding
printfs etc in CompletionThread.cpp.

The reason the rc command never returns is that rc is waiting for input on
stdin. When your buffer is unsaved emacs will write the contents of the
buffer to stdin of rc to send it the current contents. If you can reproduce
it when the file is saved that is likely better (rc shouldn't be passing
the --unsaved-file command then.

Note also that rc doesn't write anything to stdout for completion commands
(since they sometimes take a second or two to finish). What we do instead
is send them our persistent diagnostics buffer.

Long story short, to reproduce try this:

- Try to make it happen for a buffer that is saved. (even if it doesn't
compile at the time, if you're trying to complete it probably won't)
- When you're able to check rdm for what command emacs ran. (Like you did
before but hopefully this time it won't include an --unsaved-file switch)
- Start rc -m --elisp (This is where the completion output will arrive)
- Run the rc command that wasn't able to produce completions.

99% of the machinery of completions is deep inside clang so if it can't get
any candidates it might be because clang can't get any.

Let me know if this helps any.

Anders

**Comment by chep** [comment]
Hi Anders,

You said "99% of the machinery of completions is deep inside clang". I just made
a system update which includes a clang update. After rollback, completion is
working again...

I think you can close the issue, thanks for help.

**Comment by Andersbakken** [comment]
Ah. Glad to hear. Completions seem to be impacted by updates more than most
things.

Anders
